src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/ImpISQLServerBulkRecord_IssuesTest.java

```diff
@@ -0,0 +1,435 @@
+        bcOperation.writeToServer(bData);
+        bcOperation.close();
+
+        ResultSet rs = stmt.executeQuery("select * from " + destTable);
+        while (rs.next()) {
+            assertEquals(rs.getString(1), value);
+        }
+    }
+
+    /**
+     * Testing that setting out of range value for small datetime is throwing the proper message
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testSmalldatetimeOutofRange() throws Exception {
+        variation = "testSmalldatetimeOutofRange";
+        BulkDat bData = new BulkDat(variation);
+        String value = ("1954-05-22 02:44:00.0").toString();
+
+        query = "CREATE TABLE " + destTable + " (smallDATA smalldatetime)";
+        stmt.executeUpdate(query);
+
+        try {
+            SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+            bcOperation.setDestinationTableName(destTable);
+            bcOperation.writeToServer(bData);
+            bcOperation.close();
+        }
+        catch (Exception e) {
+            if (e instanceof SQLServerException) {
+                assertTrue(e.getMessage().contains("Conversion failed when converting character string to smalldatetime data type"),
+                        "Invalid Error message: " + e.toString());
+            }
+        }
+        ResultSet rs = stmt.executeQuery("select * from " + destTable);
+        while (rs.next()) {
+            assertEquals(rs.getString(1), value);
+        }
+    }
+
+    /**
+     * Test binary out of length (sending length of 6 to binary (5))
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testBinaryColumnAsByte() throws Exception {
+        variation = "testBinaryColumnAsByte";
+        BulkDat bData = new BulkDat(variation);
+        query = "CREATE TABLE " + destTable + " (col1 binary(5))";
+        stmt.executeUpdate(query);
+
+        try {
+            SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+            bcOperation.setDestinationTableName(destTable);
+            bcOperation.writeToServer(bData);
+            bcOperation.close();
+        }
+        catch (Exception e) {
+            if (e instanceof SQLServerException) {
+                assertTrue(e.getMessage().contains("The given value of type"), "Invalid Error message: " + e.toString());
+            }
+            else {
+                fail(e.getMessage());
+            }
+        }
+    }
+
+    @Test
+    public void testBinaryColumnAsString() throws Exception {
+        variation = "testBinaryColumnAsString";
+        BulkDat bData = new BulkDat(variation);
+        query = "CREATE TABLE " + destTable + " (col1 binary(5))";
+        stmt.executeUpdate(query);
+
+        try {
+            SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+            bcOperation.setDestinationTableName(destTable);
+            bcOperation.writeToServer(bData);
+            bcOperation.close();
```

Comment:
after bcOperation.close(), you should force the test to fail with something like:
`fail("BulkCopy executed when it it was expected to fail");`